src/views/payment-sheet-views/card-view.js

```diff
@@ -77,6 +71,9 @@ CardView.prototype._initialize = function () {
 };
 
 CardView.prototype._generateHostedFieldsOptions = function () {
```

Comment:
I'm having some difficulty understanding what this method does now. Not sure if it's because of the `if` nesting, or because it has a lot of special cases. Is it possible to move the special cases to the top and do an early return? Would it be an improvement to add each field option based on whitelisted conditions instead of initializing options and removing them based on blacklist conditions?

These are just suggestions, not demands :)